File: static/CWE-601/csharp/vulnerable.cs
// CWE-601: URL Redirection to Untrusted Site ('Open Redirect')
// A user-controlled URL from the query string is used as a redirect target without validation.
// Vulnerable sink: Response.Redirect(Request.QueryString["url"])

using System.Web;

namespace GuineaPig.Sast
{
    public class CWE_601
    {
        public void Redirect(HttpRequest Request, HttpResponse Response)
        {
            string url = Request.QueryString["url"];
            Response.Redirect(url);
        }
    }
}

File: static/CWE-601/javascript/vulnerable.js
// CWE-601: URL Redirection to Untrusted Site ('Open Redirect')
// Untrusted request input is used as the redirect target without validation.
// Vulnerable sink: res.redirect(req.query.url)

const express = require('express');

const app = express();

app.get('/redirect', (req, res) => {
  const url = req.query.url;
  res.redirect(url);
});

app.listen(3000);
